### ISSUE TITLE
Bug in nested context manager

### ISSUE THREAD

**Issue by TomDLT** [joblib/joblib#642]
To reproduce:
```py
from joblib import Parallel, delayed

def run():
    with Parallel(n_jobs=1) as parallel:
        assert parallel.n_jobs == 1

Parallel(n_jobs=1)(delayed(run)() for _ in range(1))
```
should not raise `AssertionError`. Instead, we have `parallel.n_jobs == -1`.

A git bisect leads to #538

**Comment by lesteve** [comment]
I can not reproduce ... which version of joblib (I guess master but just to confirm) and `platform.platform()` output?

**Comment by TomDLT** [comment]
Apparently it was solved in #602, I must have failed to update master.
Sorry for the noise
